Sources/SpeziAccount/SpeziAccount.docc/AccountService/Creating your own Account Service.md

```diff
@@ -0,0 +1,258 @@
+
+### Configuration
+
+Every account service has to provide their ``AccountService/configuration``.
+It is required to supply the ``SupportedAccountKeys``.
+Other configurations can be provided through the optional result builder closure.
+
+The code example below demonstrates how to configure an `AccountService` that supports ``SupportedAccountKeys/arbitrary`` storage of
+account keys and additionally defines an ``UserIdConfiguration`` and ``RequiredAccountKeys`` configuration.
+
+```swift
+AccountServiceConfiguration(supportedKeys: .arbitrary) {
+    UserIdConfiguration.emailAddress // userId key has an "email" label and uses the email keyboard
+
+    RequiredAccountKeys {
+        \.userId
+        \.password
+    }
+}
+```
+
+> Important: If your `AccountService` only supports to store a limited set of `AccountKey`s alongside the user account information
+    use ``SupportedAccountKeys/exactly(_:)`` to communicate the supported set of `AccountKey`s.
+    Further, refer to the <doc:Custom-Storage-Provider> article on how to integrate with an ``AccountStorageProvider``.
+
+### Specifying Identity Provider
+
+An `IdentityProvider` is an entry point to setting up an account with your `AccountService`.
+Your `AccountService` might support one or more identity provider.
+For example, a user might sign up with a userId and a password or use a Single-Sign-On provider like
+[Sign in with Apple](https://developer.apple.com/sign-in-with-apple).
+
+Use the ``IdentityProvider`` property wrapper inside your `AccountService` to provide UI components that are used
+inside the ``AccountSetup`` view that guide a user through setting up an account with your `AccountService`.
+
+> Tip: SpeziAccount provides some UI components out of the box that can be customized to your needs.
+    ``AccountSetupProviderView`` and ``SignInWithAppleButton`` are two examples.
+
+The example below demonstrates a setup with two `IdentityProvider` declarations.
+For each of them, you create a view component which interact with your `AccountService`.
+You can use the SwiftUI [`Environment`](https://developer.apple.com/documentation/swiftui/environment) to access your
+`AccountService` within your view.
+
+You can specify a ``AccountSetupSection`` and an `enabled` flag with your `IdentityProvider` declaration.
+You can dynamically change the configuration using the ``IdentityProvider/projectedValue`` of the `IdentityProvider`.
+
+
+```swift
+import AuthenticationServices
+import Spezi
+import SpeziAccount
+
+
+private struct MySetupView: View {
+    @Environment(MyAccountService.self)
+    private var service
+
+    var body: some View {
+        // An AccountSetupProviderView that doesn't support password reset functionality.
+        AccountSetupProviderView { credentials in
+            try await service.login(userId: credentials.userId, password: credentials.password)
+        } signup: { details in
+            // try await service.signup(details)
```

Comment:
Is that commented out by design?

Reply:
Not sure how the line got commented out, removed the double slashes, thanks 👍